pkg/monitoring/rules/recordingrules/operator.go

```diff
@@ -32,3 +42,19 @@ var operatorRecordingRules = []operatorrules.RecordingRule{
 		Expr:       intstr.FromString(`sum by (container, reason)(kubevirt_memory_delta_from_requested_bytes)`),
 	},
 }
+
+func buildOperatorHealthStatusExpr() intstr.IntOrString {
+	criticalExpr := fmt.Sprintf(
+		`(vector(%d) and on() ((kubevirt_hco_system_health_status==%d) or (count(ALERTS{kubernetes_operator_part_of="kubevirt", alertstate="firing", operator_health_impact="critical"})>0)))`,
+		int64(CriticalImpact), int64(metrics.SystemHealthStatusError),
+	)
+
+	warningExpr := fmt.Sprintf(
+		`(vector(%d) and on() ((kubevirt_hco_system_health_status==%d) or (count(ALERTS{kubernetes_operator_part_of="kubevirt", alertstate="firing", operator_health_impact="warning"})>0)))`,
+		int64(WarningImpact), int64(metrics.SystemHealthStatusWarning),
+	)
+
+	healthyExpr := fmt.Sprintf("vector(%d)", int64(NoImpact))
+
+	return intstr.FromString("label_replace(" + criticalExpr + " or " + warningExpr + " or " + healthyExpr + `,"name","kubevirt-hyperconverged","","")`)
```

Comment:
nit:

not so important because it happens only once, but I think that, performance wise,  there are too many concatenations here. Please check if you can use `strings.Builder` here.

Reply:
```
	var expr strings.Builder
	expr.WriteString(addHCOLabelPrefix)
	expr.WriteString(criticalExpr)
	expr.WriteString(orDivider)
	expr.WriteString(warningExpr)
	expr.WriteString(orDivider)
	expr.WriteString(healthyExpr)
	expr.WriteString(addHCOLabelSuffix)

	return intstr.FromString(expr.String())
```

It would look something like this, right?
I would suggest accepting the performance penalty (as you said happens once) for improved readability that the current approach offers

let me know your thoughts